cmd/clusterctl/client/tree/discovery.go

```diff
@@ -93,54 +104,131 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 
 		if visible {
 			if machineInfra, err := external.Get(ctx, c, &m.Spec.InfrastructureRef, cluster.Namespace); err == nil {
-				tree.Add(m, machineInfra, ObjectMetaName("MachineInfrastructure"), NoEcho(true))
+				tree.Add(m, machineInfra, ObjectMetaName("MachineInfrastructure"), NoEcho(!options.ShowAllResources))
```

Comment:
I'd like `ShowAllResources` to show all the fields and not hide the machine infra and bootstrap configs, which setting `NoEcho(true)` would do. If the user calls `clusterctl describe` with the default options, the behavior wouldn't change, but if they wanted to show all, then it would be able to toggle showing the machine infra and bootstrap config. What do you think?

Reply:
Given this is being driven by the library, I don't see a problem in requiring the GUI to set the flags and turn off echo.
Also, the two flags have different semantical meanings, and I think it is an error to think that what a GUI expect applies to other interfaces built on top of this library